xlators/meta/src/option-file.c

```diff
@@ -43,3 +43,154 @@ meta_option_file_hook(call_frame_t *frame, xlator_t *this, loc_t *loc,
 
     return 0;
 }
+
+/* adjustable variant */
+
+static volume_option_t *
+find_volopt(xlator_t *xl, char *key)
+{
+    volume_opt_list_t *vol_opt_list = NULL;
+    volume_option_t *vol_opt = NULL;
+    int i = 0;
+
+    list_for_each_entry(vol_opt_list, &xl->volume_options, list)
+    {
+        vol_opt = vol_opt_list->given_opt;
+        while (vol_opt->key[0]) {
+            if (vol_opt->fetch) {
+                for (i = 0; i < ZR_VOLUME_MAX_NUM_KEY; i++) {
+                    if (vol_opt->key[i]) {
+                        if (strcmp(vol_opt->key[i], key) == 0)
+                            return vol_opt;
+                    } else
+                        break;
+                }
+            }
+            vol_opt++;
+        }
+    }
+
+    return NULL;
+}
+
+static int
+adjustable_option_file_fill(xlator_t *this, inode_t *inode, strfd_t *strfd)
+{
+    volume_option_t *vol_opt = NULL;
+    char *optval = NULL;
+    xlator_t *xl = NULL;
+    dentry_t *dentry = NULL;
+    dentry_t *trav = NULL;
+
+    xl = meta_ctx_get(inode, this);
+    list_for_each_entry(trav, &inode->dentry_list, inode_list)
+    {
+        dentry = trav;
+        break;
+    }
+    vol_opt = find_volopt(xl, dentry->name);
+    optval = vol_opt->fetch(vol_opt, xl);
+
+    strprintf(strfd, "%s\n", optval);
+    GF_FREE(optval);
+
+    return strfd->size;
+}
+
+static int
+adjustable_option_file_write_str(xlator_t *this, fd_t *fd, char *str)
+{
+    volume_option_t *vol_opt = NULL;
+    xlator_t *xl = NULL;
+    dentry_t *dentry = NULL;
+    dentry_t *trav = NULL;
+    int ret = 0;
+    size_t len = 0;
+
+    len = strlen(str);
+    if (len == 0)
```

Comment:
I see, the current logic is bogus. Sure thing, `len` should be checked for zeroness before chomping, because to check the last character of the string, the string should contain at least one chaaracter. But there is no reason to bail out on empty input; only chomping is what should not be attempted in that case.